src/directives/formly-form.js

```diff
@@ -112,28 +112,56 @@ function formlyForm(formlyUsability, formlyWarn, $parse, formlyConfig, $interpol
     setupFields()
 
     // watch the model and evaluate watch expressions that depend on it.
-    $scope.$watch('model', onModelOrFormStateChange, true)
+    if ($scope.options.manualModelWatcher) {
+      const stringWatchers = {}
+
+      angular.forEach($scope.fields, function(field, index) {
+        if (field.data && Array.isArray(field.data.watch)) {
```

Comment:
If it's part of the core API of angular-formly, then it shouldn't go in `data`, but it could go in `extras`.